dashboard/app/util_test.go

```diff
@@ -203,6 +216,12 @@ func (c *Ctx) Close() {
 		}
 	}
 	unregisterContext(c)
+	// Verify that we did not accidentally modify the global config.
+	configAfterTest, err := json.MarshalIndent(getConfig(context.Background()), "", " ")
```

Comment:
This is good, but this may not be enough. This won't catch modifications within the test scope:
```
config.Foo = true
defer func() {
  config.Foo = false
}
```

And won't catch modifications in tests that don't use Ctx.

I am thinking we could serialize the config when it's installed and then also check in getConfig when running tests.

Reply:
I've updated the code to run the check on every `getConfig()` and after every test.